dist/apache-unomi-tracker.cjs.js

```diff
@@ -0,0 +1,1571 @@
+
+
+        wem._dispatchJSExperienceDisplayedEvents(); // Some event may not need to be send to unomi, check for them and filter them out.
+
+
+        wem._filterUnomiEvents(); // Add referrer info into digitalData.page object.
+
+
+        wem._processReferrer(); // Build view event
+
+
+        var viewEvent = wem.buildEvent('view', wem.buildTargetPage(), wem.buildSource(wem.digitalData.site.siteInfo.siteID, 'site'));
+        viewEvent.flattenedProperties = {}; // Add URLParameters
+
+        if (location.search) {
+          viewEvent.flattenedProperties['URLParameters'] = wem.convertUrlParametersToObj(location.search);
+        } // Add interests
+
+
+        if (wem.digitalData.interests) {
+          viewEvent.flattenedProperties['interests'] = wem.digitalData.interests;
+        } // Register the page view event, it's unshift because it should be the first event, this is just for logical purpose. (page view comes before perso displayed event for example)
+
+
+        wem._registerEvent(viewEvent, true);
+
+        if (wem.activateWem) {
+          wem.loadContext();
+        } else {
+          wem._executeFallback('wem is not activated on current page');
+        }
+      });
+    },
+
+    /**
+     * get the current loaded context from Unomi, will be accessible only after loadContext() have been performed
+     * @returns {object} loaded context
+     */
+    getLoadedContext: function getLoadedContext() {
+      return wem.cxs;
+    },
+
+    /**
+     * In case Unomi contains rules related to HTML forms in the current page.
+     * The logic is simple, in case a rule exists in Unomi targeting a form event within the current webpage path
+     *  - then this form will be identified as form to be watched.
+     * You can reuse this function to get the list of concerned forms in order to attach listeners automatically for those form for example
+     * (not that current tracker is doing that by default, check function: _registerListenersForTrackedConditions())
+     * @returns {string[]} form names/ids in current web page
+     */
+    getFormNamesToWatch: function getFormNamesToWatch() {
+      return wem.formNamesToWatch;
+    },
+
+    /**
+     * Get current session id
+     * @returns {null|string} get current session id
+     */
+    getSessionId: function getSessionId() {
+      return wem.sessionID;
+    },
+
+    /**
+     * This function will register a personalization
+     *
+     * @param {object} personalization the personalization object
+     * @param {object} variants the variants
+     * @param {boolean} [ajax] Deprecated: Ajax rendering is not supported anymore
+     * @param {function} [resultCallback] the callback to be executed after personalization resolved
+     * @returns {undefined}
+     */
+    registerPersonalizationObject: function registerPersonalizationObject(personalization, variants, ajax, resultCallback) {
```

Comment:
I prefer not for now, because it would break any existing call to this function.
In our own project, we may still use it with this param, and removing it here would:
- either break our code
- either force us to redefined the signature on our side
This definitely should be done, but again the code needed to be outsourced at it is for a first commit.

Then discussion will be open to see what to improve and clean